nvflare/app_opt/sklearn/joblib_model_param_persistor.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from joblib import dump, load
+
+from nvflare.apis.event_type import EventType
+from nvflare.apis.fl_constant import FLContextKey
+from nvflare.apis.fl_context import FLContext
+from nvflare.app_common.abstract.learnable_persistor import LearnablePersistor
+from nvflare.app_common.abstract.model import ModelLearnable, ModelLearnableKey, make_model_learnable
+from nvflare.app_common.app_constant import AppConstants
+
+
+class JoblibModelParamPersistor(LearnablePersistor):
```

Comment:
Sorry, by reviewing all the examples again, I know why you were using ModelPersistor.

Our ScatterAndGather controller now assume things need to be a "ModelLearnable".

I think we need a broader discussion on whether (1) ModelLearnable means deep learning models
(2) Could SAG only apply on deep learning models.

Could you modify this back to use ModelPersistor?

Instead of calling "JoblibModelParamPersistor", I prefer "SKLearnModelPersistor" or something similar.
I don't know what joblib is, but I do understand this persistor is for scikit-learn models/params when I see the latter name.

Reply:
sounds good, we can discuss more. As for joblib, it is an alternative / extension of pickle, used by sklearn according to its guideline, so I think we should keep the name and potentially move out of sklearn folder together with the data_loader function.